src/prommis/nanofiltration/costing/diafiltration_cost_model.py

```diff
@@ -397,12 +348,73 @@ def capital_cost_constraint(blk):
-            return blk.fixed_operating_cost == units.convert(
-                blk.pump_power
-                * blk.costing_package.electricity_cost
-                / blk.costing_package.operating_time,  # per one year
-                to_units=blk.costing_package.base_currency
-                / blk.costing_package.base_period,
+        if OPEX:
+            # calculate the pump head: pump Ref [1] Eqn 1.1
+            blk.pump_head = Var(
+                initialize=10,
+                doc="Pump head in meters",
+                units=units.m,
             )
+
+            @blk.Constraint()
+            def pump_head_equation(blk):
+                return blk.pump_head == units.convert(
+                    (
+                        outlet_pressure
+                        * blk.pump_head_factor
+                        / blk.costing_package.specific_gravity
+                    ),
+                    to_units=units.m,
+                )
+
+            # calculate the pump power: pump Ref [2] Eqn 7
+            blk.pump_power = Var(
+                initialize=10,
+                doc="Pump power in kWh required for the operational period",
+                units=units.kWh,
+            )
+
+            grav_constant = units.convert(
+                Constants.acceleration_gravity, to_units=units.m / units.hr**2
+            )
+
+            @blk.Constraint()
+            def pump_power_equation(blk):
+                return blk.pump_power == units.convert(
+                    (
+                        units.convert(
+                            (
+                                inlet_vol_flow
+                                * blk.costing_package.density
+                                * grav_constant
+                                * blk.pump_head
+                                / blk.pump_power_factor
+                                / blk.pump_efficiency
+                            ),
+                            to_units=units.kW,
+                        )
+                        * blk.costing_package.operating_time  # per one year
+                    ),
+                    to_units=units.kWh,
+                )
+
+            @blk.Constraint()
+            def fixed_operating_cost_constraint(blk):
+                return blk.fixed_operating_cost == units.convert(
+                    blk.pump_power
+                    * blk.costing_package.electricity_cost
+                    / blk.costing_package.operating_time,  # per one year
+                    to_units=blk.costing_package.base_currency
+                    / blk.costing_package.base_period,
+                )
+
+        else:
+
+            @blk.Constraint()
+            def fixed_operating_cost_constraint(blk):
+                return (
+                    blk.fixed_operating_cost
+                    == 0
+                    * blk.costing_package.base_currency
+                    / blk.costing_package.base_period
```

Comment:
In the past, Pyomo has encountered issues building expressions as 0 * [unit container]. It may be safer to remove the constraint and just fix `fixed_operating_cost` to 0 in this case. Or, set the initial value to 0 when creating the variable so we don't need the `else` case.

Reply:
@bpaul4 @andrewlee94 I am having a DOF issue while trying to implement these changes. When I initialize the `blk.fixed_operating_cost` variable at 0 and fix it (to eliminate the need for `else`) my DOF of the cost block is now -1.

Reply:
You should dig into that further to see what is going on. I would start with the Diagnostics Toolbox and see what it says about structural singularities.

Reply:
@andrewlee94 Thanks. I didn't initially see anything, and was surprised to see that initializing `blk.fixed_operating_cost` at 0, but not fixing the variable results in DOF=1. I'll dig further into the Toolbox.